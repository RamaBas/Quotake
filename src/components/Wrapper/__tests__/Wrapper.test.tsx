import { render, screen } from '@testing-library/react';
import AppWrapper from '../Wrapper';

// Mock de los componentes hijos para simplificar los snapshots
jest.mock('../../AppHead/AppHead', () => () => <div data-testid="app-head-mock" />);
jest.mock('../../AddPhrases/AddPhrasesButton', () => () => <div data-testid="add-phrase-button-mock" />);

describe('AppWrapper Component', () => {
  it('renders correctly with main structure', () => {
    const { container, getByTestId } = render(
      <AppWrapper>
        <div>Child Content</div>
      </AppWrapper>
    );
    
    expect(getByTestId('app-head-mock')).toBeInTheDocument();
    expect(getByTestId('add-phrase-button-mock')).toBeInTheDocument();
    
    expect(screen.getByText('Child Content')).toBeInTheDocument();
    
    // Snapshot del wrapper completo
    expect(container.firstChild).toMatchSnapshot();
  });
});