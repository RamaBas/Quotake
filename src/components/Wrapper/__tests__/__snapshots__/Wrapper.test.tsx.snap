// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppWrapper Component renders correctly with main structure 1`] = `
<div
  class="MuiBox-root css-17rx1af"
>
  <div
    data-testid="app-head-mock"
  />
  <main
    class="MuiBox-root css-cpcsu7"
  >
    <div>
      Child Content
    </div>
  </main>
  <div
    data-testid="add-phrase-button-mock"
  />
</div>
`;
